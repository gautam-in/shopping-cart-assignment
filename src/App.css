body{
    font-family: 'Dosis', sans-serif;;
    /* padding: 20px 60px; */
}

a {
    text-decoration: none;
    color: black;
    cursor: pointer;
}

* {
    box-sizing: border-box;
}

