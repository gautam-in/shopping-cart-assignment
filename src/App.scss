@import "./config";

.main {
  max-width: 1024px;
  margin: 5.81rem auto 0 auto;
}

@media (max-width: 732px) {
  .main {
    margin: 4rem auto 0 auto;
    overflow: hidden;
  }
}
