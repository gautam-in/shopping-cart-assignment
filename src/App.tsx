import React, { useReducer } from "react";
import { Routes, Route } from "react-router-dom";
import { Layout } from "./components/Layout/Layout";
import { ProductGrid } from "./components/ProductGrid/ProductGrid";
import { Home } from "./pages/Home/Home";
import { Products } from "./pages/Products/Products";
import { Register } from "./pages/Register/Register";
import { SignIn } from "./pages/SignIn/SignIn";
import { PageNotFound } from "./pages/PageNotFound/PageNotFound";
import { Cart } from "./pages/Cart/Cart";
import { cartReducer } from "./utils/cartReducer";
import type { Product } from "./types/customTypes";

const initialCartState: Product[] = [] ;

export const App = () => {

  const [cartState, cartDispatch] = useReducer(cartReducer, initialCartState);

    return (
        <Routes>
          <Route path="/" element={<Layout size={cartState.length}/>}>
            <Route index element={<Home />}/>  
            <Route path="products" element={<Products />}>
              <Route path="all" element={<ProductGrid cartDispatch={cartDispatch} />}/>
              <Route path=":id" element={<ProductGrid cartDispatch={cartDispatch} />}/>
            </Route> 
            <Route path="signin" element={<SignIn />} />
            <Route path="register" element={<Register />}/>
            <Route path="cart" element={<Cart cartState={cartState} cartDispatch={cartDispatch}/>}/>
          </Route>
          <Route path="*" element={<PageNotFound />}/>   
        </Routes>
      )
}