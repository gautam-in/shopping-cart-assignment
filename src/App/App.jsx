import React from 'react';
import { Router, Route, Switch, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

import { history } from '../_helpers';
import { alertActions } from '../_actions';
import { PrivateRoute } from '../_components';
import { HomePage } from '../HomePage';
import { LoginPage } from '../LoginPage';
import { RegisterPage } from '../RegisterPage';

class App extends React.Component {
    constructor(props) {
        super(props);

        history.listen((location, action) => {
            // clear alert on location change
            this.props.clearAlerts();
        });
    }

    render() {
        const { alert } = this.props;
        return (
            <div className="jumbotron">
                <div className="container">
                    <div className="col-sm-8 col-sm-offset-2">
                        
                        <Router history={history}>
                            <Switch>
                                <PrivateRoute exact path="/" component={HomePage} />
                                <Route path="/login" component={LoginPage} />
                                <Route path="/register" component={RegisterPage} />
                                <Redirect from="*" to="/" />
                            </Switch>
                        </Router>
                    </div>
                </div>
                {alert.message &&
                            <p style={{color:'red',paddingLeft:'62%'}}>{alert.message}</p>
                }
            </div>
        );
    }
}

function mapState(state) {
    const { alert } = state;
    return { alert };
}

const actionCreators = {
    clearAlerts: alertActions.clear
};

const connectedApp = connect(mapState, actionCreators)(App);
export { connectedApp as App };