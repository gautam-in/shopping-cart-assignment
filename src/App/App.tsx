import { Route, Routes } from "react-router";
import { FooterComponent } from "../BaseComponent/Footer/FooterComponent";
import { HeaderComponent } from "../BaseComponent/Header/HeaderComponent";
import { RegisterComponent } from "../Pages/Register/RegisterComponent";
import { SignInComponent } from "../Pages/SignIn/SignInComponent";
import HomeComponent from "../Pages/Home/HomeComponent";
import ProductComponent from "../Pages/Product/ProductComponent";
import CartPageComponent from "../Pages/Cart/CartPageComponent";
import { loadBanners, loadCategories, loadProduct } from "../Model/Model";
import { useEffect } from "react";

export const App: React.FC<{}> = () => {
    useEffect(() => {
        loadBanners();
        loadCategories();
        loadProduct();
    }, []);
    return (
        <>
            <HeaderComponent />
            <Routes>
                <Route path="/" element={<HomeComponent />} />
                <Route path="signin" element={<SignInComponent />} />
                <Route path="register" element={<RegisterComponent />} />
                <Route path="product" element={<ProductComponent />} />
                <Route path="cart" element={<CartPageComponent />} />
            </Routes>
            <FooterComponent />
        </>
    );
};
