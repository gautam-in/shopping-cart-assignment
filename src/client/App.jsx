import React from 'react';
import { renderRoutes } from 'react-router-config';
import withStyles from 'isomorphic-style-loader/withStyles';

import Layout from './components/Layout';
import styles from './assets/styles/styles.scss';

const App = ({ route }) => (
  <Layout>
    {renderRoutes(route.routes)}
  </Layout>
);

export default {
  component: withStyles(styles)(App),
  // loadData: ({ dispatch }) => dispatch(fetchCurrentUser())
};
