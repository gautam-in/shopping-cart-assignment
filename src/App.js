import './App.css';
// import {  HomePage, LoginPage, ProductListingPage, SignupPage } from './pages/index-page';
import { Route, Routes, useLocation } from 'react-router-dom';
import React, { Suspense, lazy } from 'react';
import Loader from './components/loading/Loader';

const CartPage = lazy(() => import("./pages/CartPage"))
const HomePage = lazy(() => import("./pages/HomePage"))
const LoginPage = lazy(() => import("./pages/LoginPage"))
const ProductListingPage = lazy(() => import("./pages/ProductListingPage"))
const SignupPage = lazy(() => import("./pages/SignupPage"))

export const LocationDisplay = () => {
  const location = useLocation()

  return <div data-testid="location-display">{location.pathname}</div>
}
function App() {

  return (
    <Suspense fallback = {<Loader type = {"spin"} color = {"blue"} className = {"loader-align"}/>}>
      <div className="App" data-testid = "app">
        <Routes>
          <Route path = "/" element = {<HomePage />} />
          <Route path = "login" element = {<LoginPage />} />
          <Route path = "signup" element = {<SignupPage />} />
          <Route path = "cart" element = {<CartPage />} />
          <Route path = "products" element = {<ProductListingPage />}>
            <Route path = "cart" element = {<CartPage />} />
          </Route>
        </Routes>
        <LocationDisplay />

      </div>
    </Suspense>
  );
}

export default App;
