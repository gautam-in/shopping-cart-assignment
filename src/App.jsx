import React, { lazy, Suspense } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import "./App.scss";
import ErrorBoundary from "./Component/ErrorBoundary/errorBoundary.component";
import Footer from "./Component/Footer/footer.component";
import Header from "./Component/Header/header.component";
import Spinner from "./Component/Spinner/spinner.component";
import CartContainer from "./Screen/Cart/cart.container";

const ProductPage = lazy(() => import("./Screen/Product/product.container"));
const LoginPage = lazy(() => import("./Screen/Login/login.container"));
const HomePage = lazy(() => import("./Screen/Home/home.container"));
const RegisterPage = lazy(() => import("./Screen/Register/register.conatiner"));

function App() {
  return (
    <Router>
      <div className="app-container">
        <Header />
        <Switch>
          <ErrorBoundary>
            <Suspense fallback={<Spinner />}>
              <Route path="/" exact component={HomePage} />
              <Route path="/product" component={ProductPage} />
              <Route path="/login" exact component={LoginPage} />
              <Route path="/register" exact component={RegisterPage} />
            </Suspense>
          </ErrorBoundary>
        </Switch>
        <CartContainer />
        <Footer />
      </div>
    </Router>
  );
}

export default App;
