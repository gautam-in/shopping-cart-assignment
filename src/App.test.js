import { render } from '@testing-library/react';
import React from "react";
import App from './App';
import { act } from 'react-dom/test-utils';
import { BrowserRouter } from 'react-router-dom';

test('renders the landing page', async () => {
    await act( async () => render(
        <BrowserRouter>
            <App/>
        </BrowserRouter>
    ));
});