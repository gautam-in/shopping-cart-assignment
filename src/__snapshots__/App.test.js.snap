// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
<div
  class="App"
  data-testid="app"
>
  <div
    class="bg-color"
  >
    <header
      class="header-nav-wrapper bg-color box-shadow"
    >
      <nav
        class="navbar-wrapper"
      >
        <div
          class="navbar-inner-first"
        >
          <a
            href="/"
          >
            <img
              alt="sabka bazar logo"
              class="navbar-logo"
              sizes="40vw"
              src="logo.png"
              srcset="logo.png 480w, logo_2x.png 1080w"
            />
          </a>
          <div
            class="navbar-first-links"
          >
            <a
              aria-current="page"
              class="nav-links active"
              href="/"
            >
              Home
            </a>
            <a
              class="nav-links"
              href="/products"
            >
              Products
            </a>
          </div>
        </div>
        <div
          class="navbar-inner-second"
        >
          <div
            class="navbar-authlinks-wrapper"
          >
            <a
              class="nav-links"
              href="/login"
            >
              SignIn
            </a>
            <a
              class="nav-links"
              href="/signup"
            >
              Register
            </a>
          </div>
          <a
            class="nav-cart-items"
            data-testid="cart-link"
            href="/cart"
          >
            Cart:
            <span>
              <b>
                0  item
              </b>
            </span>
          </a>
        </div>
      </nav>
    </header>
    <main
      class="bg-color"
      data-testid="home-page"
    >
      <div
        class="home-banner-wrapper box-shadow"
      >
        <div>
          <picture>
            <source
              class="home-banner-image"
              media="(min-width: 1200px)"
              srcset="undefined"
            />
            <source
              class="home-banner-image"
              media="(max-width: 1200px)"
              srcset="undefined"
            />
            <img
              alt="offer banner"
              class="home-banner-image"
              src="undefined"
            />
          </picture>
        </div>
        <i
          class="fa-solid fa-angle-left banner-left-arrow"
          data-testid="left-slider"
        />
        <i
          class="fa-solid fa-angle-right banner-right-arrow"
          data-testid="right-slider"
        />
      </div>
      <div
        class="flex home-cat-wrapper"
      />
    </main>
    <footer
      class="footer-wrapper"
      data-testid="footer"
    >
      <p>
        Copyright Â© 2011-2018 Sabka Bazaar Grocery Supplies Pvt Ltd
      </p>
    </footer>
  </div>
  <div
    data-testid="location-display"
  >
    /
  </div>
</div>
`;
